### PR TITLE
learn basic javascript

### PR DESCRIPTION
**Pull request by adi061001**

### CODE REVIEW

File: js-basic/Callback/callback.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+function fun() {
+  console.log('the exam is done ');
+}
+function fun2(fun) {
+  console.log('result is out ');
+  fun();
+}
```
Comment (Ritikthoughtwin) [inline js-basic/Callback/callback.js]: Naming conventions should be  proper and informative.

File: js-basic/Closure/1-clos.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+function App() {
+  let num1 = 23;
+  function num() {
+    console.log(num1);
+  }
+  return num;
+}
+App()();
+//////neested function store date after destroy for child function
+
```
Comment (Ritikthoughtwin) [inline js-basic/Closure/1-clos.js]: add comments only with double slash what is the sense of using multiple slashes?

File: js-basic/Array/array.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// a.unshift(2);  it will add on start chnage in orginal array
+
+// a.pop(); it will remove last element odf array
+
+// let b = a.concat(k); add two more than array in new array it will not change orignal array
+
+// a.splice(1, 2,4);callback values index deltecount addnum
+
+// let n = a.slice(0, 2);  it will not chnage and original array it will take out value from and array
+
```
Comment (Ritikthoughtwin) [inline js-basic/Array/array.js]: Solve these methods and add results and examples as well

File: js-basic/hoisting/host.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+///////////////////////////////HOISTING
```
Comment (Ritikthoughtwin) [inline js-basic/hoisting/host.js]: No need to create headings like this, add double-slash comments

File: js-basic/hoisting/host.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+//////////////////////////////////////////////////////////////////////Error
+// fun();
+// var fun = () => {
+//   console.log('track');
+// };
+//////////////////////////////////////arrow funactin is not work with hositing
+
+a = [1, 4, 5, 6, 7, 3];
+fun(...a);
+a.splice(0, 1);
```
Comment (Ritikthoughtwin) [inline js-basic/hoisting/host.js]: Fix naming conventions.

File: js-basic/hoisting/hosting.html [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Document</title>
+  </head>
+  <body>
+    <script>
+      let x = 23;
+      function App() {
```
Comment (Ritikthoughtwin) [inline js-basic/hoisting/hosting.html]: match the spelling of folder and file name.

File: js-basic/hoisting/hosting.html [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+        console.log('aditya shrama ');
+      }
+      console.log(App);
+    </script>
+  </body>
+</html>
```
Comment (Ritikthoughtwin) [inline js-basic/hoisting/hosting.html]: Why did you created an HTML file here?

File: js-basic/Callback/callback.js (hunk 1 above)
Comment (adi061001) [reply js-basic/Callback/callback.js]: all variable names are change into informative

File: js-basic/Closure/1-clos.js (hunk 2 above)
Comment (adi061001) [reply js-basic/Closure/1-clos.js]: removed extra slash

File: js-basic/hoisting/hosting.html (hunk 7 above)
Comment (adi061001) [reply js-basic/hoisting/hosting.html]: checking somthing that's why there is html file but i removed file

File: js-basic/hoisting/host.js (hunk 5 above)
Comment (adi061001) [reply js-basic/hoisting/host.js]: all name are change in format

File: js-basic/Array/array.js (hunk 3 above)
Comment (adi061001) [reply js-basic/Array/array.js]: all the method add & result

**Comment by Ritikthoughtwin** [comment]
Fix commit messages

File: js-basic/Array/array.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
+let a = [18, 23, 23, 4, 4, 5];
+let k = ['Aditya', 'vinod', 'rakesh', 'crab'];
```
Comment (tw-vineet) [inline js-basic/Array/array.js]: Variables name should be meaningful an informative check and change
ex:- let nameArr = ['Aditya', 'vinod', 'rakesh', 'crab'];

File: js-basic/Closure/1-clos.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+function App() {
```
Comment (tw-vineet) [inline js-basic/Closure/1-clos.js]: Maintain naming convention we always declare function in camelcase ex:- app

File: js-basic/Array/array.js (hunk 8 above)
Comment (adi061001) [reply js-basic/Array/array.js]: Already done

File: js-basic/Closure/1-clos.js (hunk 9 above)
Comment (adi061001) [reply js-basic/Closure/1-clos.js]: getting issue to push so some old file are also  pushed

File: OSR_3-Javascript/Array.method/array.js [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+//   return fulldetails;
+// });
+// console.log(newdata);
+
+// function test() {
+//   return name;
+//   function name() {}
+// }
+// console.log(test());
+
```
Comment (Ritikthoughtwin) [inline OSR_3-Javascript/Array.method/array.js]: Uncomment all this code.

File: OSR_3-Javascript/Async/async_await.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+// async function fetchData() {
+//   try {
+//     let response = await fetch('https://dummyjson.com/users');
+//     let data = await response.json();
+//     console.log(data);
+//   } catch (error) {
+//     console.error('Data not get ');
+//   }
+// }
```
Comment (Ritikthoughtwin) [inline OSR_3-Javascript/Async/async_await.js]: Refer previous comment

File: OSR_3-Javascript/Promisse/Prmoisses.js [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+// });
+
+// Promise.race([promiseMethodFirst, promiseMethodSecond ])
+//   .then((res) => {
+//     console.log(res);
+//   })
+//   .catch((err) => {
+//     console.log(err);
+//   });
+
```
Comment (Ritikthoughtwin) [inline OSR_3-Javascript/Promisse/Prmoisses.js]: Refer previous.

File: OSR_3-Javascript/hoisting/host.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+//////////////////////////////////function hosting
+
+// App();
+// function App() {
+//   console.log('Aditya');
+// }
+
+// function is declared after the calling
+
```
Comment (Ritikthoughtwin) [inline OSR_3-Javascript/hoisting/host.js]: Refer previous comment

File: OSR_3-Javascript/Array.method/array.js (hunk 10 above)
Comment (adi061001) [reply OSR_3-Javascript/Array.method/array.js]: all the comments are removed

File: OSR_3-Javascript/Async/async_await.js (hunk 11 above)
Comment (adi061001) [reply OSR_3-Javascript/Async/async_await.js]: done

File: OSR_3-Javascript/hoisting/host.js (hunk 13 above)
Comment (adi061001) [reply OSR_3-Javascript/hoisting/host.js]: done

File: OSR_3-Javascript/Promisse/Prmoisses.js (hunk 12 above)
Comment (adi061001) [reply OSR_3-Javascript/Promisse/Prmoisses.js]: done

**Comment by Ritikthoughtwin** [comment]
@adi061001 Fix file names.

File: OSR_3-Javascript/Array.method/array.js [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+//Join
+// the join method return value in string
+// the join method assign value what ever we give in inside the (*) and it will put between all the elemnets
+result = a.join('*');
+
+// flat( method of array not change orignal array )
+// it open array inside array into sinle array
+let newarrayfirst = arrNamesecond.flat();
+console.log(newarrayfirst);
+
```
Comment (Ritikthoughtwin) [inline OSR_3-Javascript/Array.method/array.js:104]: use camelCase veriale names

Comment (adi061001) [reply OSR_3-Javascript/Array.method/array.js:104]: ohk done sir

**Review by Ritikthoughtwin — APPROVED** [review]

**Pull request merged**